packages/ember-engines/lib/engine-addon.js

```diff
@@ -641,6 +641,11 @@ function buildEngine(options) {
       return engineRoutesTree;
     };
 
+    // Replace `treeForApp` so no engine files leak into app namespace
+    // This prevents accidental usage of engine components/helpers/utilities
+    // inside of a host app
+    this.treeForApp = function() {};
```

Comment:
@void-mAlex could you please add one test for it? We have the node tests here where you can cover this fix - https://github.com/ember-engines/ember-engines/blob/master/packages/ember-engines/node-tests/acceptance/build-test.js